src/Components/Scorecard.js

```diff
@@ -1,150 +1,114 @@
 import { Errors } from '../Contexts/ErrorsProvider'
 import Player from './Player'
 import { scoreAHole, endRound } from '../apiCalls'
 import {useNavigate} from 'react-router-dom'
 
 const Scorecard = () => {
   const { scorecard, setScorecard } = useContext(ScorecardInfo)
-  const [ currentHole, setCurrentHole ] = useState({})
+  const [ currentHole, setCurrentHole ] = useState({par: 3, number: 1, distance: 0})
   const {errorMessage, setErrorMessage} = useContext(Errors)
   const [inProgress, setInProgress] = useState(true)
+  const [scores, setScores] = useState([])
+  const {courseInfo, setCourseInfo} = useContext(CourseInfo)
   const navigate = useNavigate()
-
-
-  const currentPlayers = () => {
-    const result = {}
-    scorecard.players.forEach(player => {
-      result[player.id] = {
-        name: player.name,
-        id: player.id,
-        score: scorecard.layout.holes[0].par,
-        totalScore: 0
-      }
-    })
-    return result
-  }
-
-  const changeScore = (id, score) => {
-    currentHole.players[id].score = score
-    setCurrentHole(currentHole)
-  }
-
+  
+  
   useEffect(() => {
     setCurrentHole({
       number: 1,
-      players: currentPlayers(),
-      par: scorecard.layout.holes.find(hole => hole.hole_number === 1).par,
-      distance: scorecard.layout.holes.find(hole => hole.hole_number === 1).distance
+      par: courseInfo.currentCourse.layout.holes.find(hole => hole.hole_number === 1).par,
+      distance: courseInfo.currentCourse.layout.holes.find(hole => hole.hole_number === 1).distance
     })
-  }, [])
-
+  }, []) 
+  
   const displayPlayers = () => {
-    return scorecard.players.map(player => {
+    scorecard.sort((a,b) => a.id - b.id)
+    return scorecard.map(player => {
       return (
         <Player
           name={player.name}
           key={player.id}
           id={player.id}
-          par={scorecard.layout.holes.find(hole => {
+          par={courseInfo.currentCourse.layout.holes.find(hole => {
             return hole.hole_number === currentHole.number
           }).par}
-          totalScore={currentHole.players[`${player.id}`].totalScore}
+          totalScore={findTotalScore(player.id)}
           changeScore={changeScore}
         />
       )
     })
   }
 
-  const nextHole = (prevHoleScores) => {
-    const hole = scorecard.layout.holes.find(hole => hole.hole_number === currentHole.number + 1)
-    if(hole.hole_number === scorecard.layout.holes.length){
+  const findTotalScore = (id) => {
+    let playerScores = scores.filter(score => score.player_id === id) 
+      return playerScores.length > 0 ? playerScores.reduce((acc, player) => {
+        acc += player.score
+        return acc
+      }, 0) : 0
+  }
+
+  const checkLastHole = () => {
+    return inProgress ?
+      <button className='next-hole-btn' onClick={postHoleScores}>NEXT HOLE</button> :
+      <button className='end-round-btn' onClick={postHoleScores}>END ROUND</button>
```

Comment:
These buttons now call the same function. We could possibly just render different text in the buttons to shorten up our code.